Assets/Plugins/CountlySDK/Services/InitializationCountlyService.cs

```diff
@@ -23,37 +29,31 @@ internal InitializationCountlyService(SessionCountlyService sessionCountlyServic
         /// <param name="serverUrl"></param>
         /// <param name="appKey"></param>
         /// <param name="deviceId"></param>
-        internal void Begin(string serverUrl, string appKey)
+        internal async Task OnInitializationComplete()
         {
-            AppKey = appKey;
-            ServerUrl = serverUrl;
-        }
+            AppKey = _configModel.AppKey;
+            ServerUrl = _configModel.ServerUrl;
 
-        /// <summary>
-        ///     Initializes the Countly SDK with default values
-        /// </summary>
-        /// <param name="salt"></param>
-        /// <param name="enablePost"></param>
-        /// <param name="enableConsoleErrorLogging"></param>
-        /// <param name="ignoreSessionCooldown"></param>
-        /// <returns></returns>
-        internal async Task SetDefaults(CountlyConfiguration configModel)
-        {
-            if (!configModel.EnableManualSessionHandling)
+            /* If location is disabled in init
+				 and no session consent is given. Send empty location as separate request.*/
+            if (_locationService.IsLocationDisabled)
```

Comment:
Why are you moving location logic into a unrelated class? Move it to the location module.